src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
```

Comment:
The gateway is SSA context, this is OpenConext. Logging PII already can happen in an OpenConext platform (e.g. in EngineBlock), not sure if that in general would be an issue. I do agree that it should be avoided though. What the NameID does or does not contain should not be a matter as it is an identifier that potentially contains PII which we cannot control.

As to how to log this, make a custom exception with a factory method that solves that, as creating a specific exception message for a specific value arguably is not really a concern that should be tackled here.